### ISSUE TITLE
`sys` is not imported in IPython 7.19.0 

### ISSUE THREAD

**Issue by frozflame** [ipython/ipython#12797]
`sys` is not imported in IPython 7.19.0:

 ```python
File "/usr/local/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py", line 547, in interact
    code = self.prompt_for_code()
  File "/usr/local/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py", line 473, in prompt_for_code
    text = self.pt_app.prompt(
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/shortcuts/prompt.py", line 1013, in prompt
    return self.app.run(set_exception_handler=set_exception_handler)
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 814, in run
    return loop.run_until_complete(
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 781, in run_async
    return await _run_async2()
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 769, in _run_async2
    await self.cancel_and_wait_for_background_tasks()
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 870, in cancel_and_wait_for_background_tasks
    await task
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/buffer.py", line 1854, in new_coroutine
    await coroutine(*a, **kw)
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/buffer.py", line 1683, in async_completer
    async for completion in self.completer.get_completions_async(
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/completion/base.py", line 269, in get_completions_async
    async for completion in completer.get_completions_async(
  File "/usr/local/lib/python3.8/site-packages/prompt_toolkit/completion/base.py", line 196, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "/usr/local/lib/python3.8/site-packages/IPython/terminal/ptutils.py", line 116, in get_completions
    exc_type, exc_value, exc_tb = sys.exc_info()
NameError: name 'sys' is not defined 
```

**Comment by Carreau** [comment]
Already fixed on 7.x. Please search recently closed issues.

**Comment by pydo** [comment]
> Already fixed on 7.x. Please search recently closed issues.

Given that there's only been 1 new release since 7.19, I assume you mean it was fixed in 7.20?

There doesn't appear to be a `CHANGELOG` in the release page or in the git repo. 
https://github.com/ipython/ipython/releases/tag/7.20.0

I found this https://github.com/ipython/ipython/blob/master/docs/source/whatsnew/version7.rst#ipython-720, but it only seems to include "highlights" from the release.

Honest question, are we expected to grep through git commits to generate an exhaustive changelog?

Thanks

**Comment by Carreau** [comment]
Yes, it's in 7.20. Don't look at github releases, they can't be deactivated but don't reflect how things works.

The [7.20](https://github.com/ipython/ipython/milestone/82?closed=1) milestone list all issues and PRs closed during 7.20.

You can also [search issues and PR by status and milestone](https://github.com/ipython/ipython/issues?q=is%3Aissue+milestone%3A7.20+is%3Aclosed) the 4th of which is "sys.excinfo() but sys is not imported
#12784 by guy4261 was closed 15 days ago milestone: 7.20"

Unlike a static changelog, using milestones also allow to change issues milestone (like if I forgot to close them or another one is opened), even after releases., and/or if an issue was thought-to-be-fixed-but-not-completely to remove it from the list of fixed issue. 

The above information should be available [on our docs](https://ipython.readthedocs.io/en/stable/whatsnew/github-stats-7.html), though; we stopped posting the same link on each release

**Comment by pydo** [comment]
Thank you for taking the time to clearly explain the release process :). I wasn't really familiar with GH milestones, but I can now see the advantages over a static `CHANGELOG` file.
